### PR TITLE
[onert] Write ChromeTracingObserver to three files by EventWriter

### PR DESCRIPTION
**Pull request by YongseopKim**
- Introduce new class EventWriter which takes EventRecorder and writes
its Records to files
- Now EventRecorder is a kind of data class to be recorded

ONE-DCO-1.0-Signed-off-by: Yongseop Kim <yons.kim@samsung.com>

### CODE REVIEW

File: runtime/onert/core/src/util/EventRecorder.cc [hunk 1]
```diff
@@ -412,146 +29,3 @@ void EventRecorder::emit(const CounterEvent &evt)
 
   _counter_events.push_back(evt);
 }
-
-void EventRecorder::writeToFile(std::ostream &os)
-{
-  std::lock_guard<std::mutex> lock{_mu};
```
Comment (YongseopKim) [inline runtime/onert/core/src/util/EventRecorder.cc:418]: I wonder why this is necessary.

**Comment by YongseopKim** [comment]
For https://github.com/Samsung/ONE/issues/3578

**Comment by YongseopKim** [comment]
@wateret could you review this?

I think it's not bad design but I'd like to get your opinion. (This is could be merged if new change is not necessary)

**Comment by wateret** [comment]
Looks better than before 😃

**Comment by YongseopKim** [comment]
@wateret okay. I change this to be reviewed officially.

**Comment by YongseopKim** [comment]
PTAL @chunseoklee and @wateret

**Review by wateret — APPROVED** [review]
LGTM

File: runtime/onert/core/src/util/EventWriter.cc [hunk 2]
```diff
@@ -0,0 +1,555 @@
...
+
+EventWriter::EventWriter(const EventRecorder &recorder) : _recorder(recorder)
+{
+  // DO NOTHING
+}
+
+void EventWriter::writeToFiles(const std::string &base_filepath)
+{
+  writeToFile(base_filepath + ".chrome.json", WriteFormat::CHROME_TRACING);
+  writeToFile(base_filepath + ".snpe.json", WriteFormat::SNPE_BENCHMARK);
```
Comment (chunseoklee) [inline runtime/onert/core/src/util/EventWriter.cc:411]:
```suggestion
  writeToFile(base_filepath, WriteFormat::SNPE_BENCHMARK);
```

Comment (chunseoklee) [reply runtime/onert/core/src/util/EventWriter.cc:411]: trace.json is used on sdk tool part :(

Comment (YongseopKim) [reply runtime/onert/core/src/util/EventWriter.cc:411]: I'll fix it soon

**Comment by YongseopKim** [comment]
I've found a bug the only on release build mode. I'll fix and then push a new patch.

File: runtime/onert/core/src/util/EventWriter.cc [hunk 3]
```diff
@@ -0,0 +1,574 @@
...
+
+EventWriter::EventWriter(const EventRecorder &recorder) : _recorder(recorder)
+{
+  // DO NOTHING
+}
+
+void EventWriter::writeToFiles(const std::string &base_filepath)
+{
+  // Note. According to an internal issue, let snpe json as just file name not '.snpe.json'
+  writeToFile(base_filepath, WriteFormat::SNPE_BENCHMARK);
```
Comment (YongseopKim) [inline runtime/onert/core/src/util/EventWriter.cc:429]: @chunseoklee

**Comment by YongseopKim** [comment]
PTAL @chunseoklee and @wateret

**Review by wateret — APPROVED** [review]
LGTM

**Review by chunseoklee — APPROVED** [review]

**Pull request merged**